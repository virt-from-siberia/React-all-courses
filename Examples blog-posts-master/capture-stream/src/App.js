import { VideoSelector, AudioSelector, Recorder, Result } from 'components'
import { createStore } from 'utils/createStore'
import './App.scss'

const store = {
  state: {
    audio: '',
    video: '',
    result: ''
  },
  setters: {
    setAudio: (_, audio) => ({ audio }),
    setVideo: (_, video) => ({ video }),
    setResult: (_, result) => ({ result })
  }
}

export const [Provider, useStore, useSetters] = createStore(store)

function App() {
  return (
    <Provider>
      <div className='container common'>
        <VideoSelector />
        <AudioSelector />
      </div>
      <Recorder />
      <Result />
    </Provider>
  )
}

export default App
