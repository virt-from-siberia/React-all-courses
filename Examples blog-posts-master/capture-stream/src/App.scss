$primary: #0275d8;
$success: #5cb85c;
$warning: #f0ad4e;
$danger: #d9534f;
$light: #f7f7f7;
$dark: #292b2c;

@mixin reset($font-family, $font-size, $color) {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  @if $font-family {
    font-family: $font-family;
  }
  @if $font-size {
    font-size: $font-size;
  }
  @if $color {
    color: $color;
  }
}

@mixin flex-center($column: false) {
  display: flex;
  justify-content: center;
  align-items: center;

  @if $column {
    & {
      flex-direction: column;
    }
  }
}

@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@200;400;600&display=swap');

*,
*::before,
*::after {
  @include reset('Montserrat', 1rem, $dark);
}

.selector {
  margin: 0.5rem;

  label {
    display: block;
    margin: 0.25rem 0;
  }
}

.container {
  margin: 0.25rem;
  @include flex-center(true);

  &.common,
  &.recorder {
    flex-direction: row;
    align-items: flex-start;
  }

  &.result {
    margin: 1rem 0;

    a {
      margin-top: 0.5rem;
      text-decoration: none;
      color: $primary;
      border-bottom: 1px dashed $primary;
    }
  }

  h2 {
    font-size: 1.4rem;
    margin: 0.25rem 0;
  }
}

.item {
  @include flex-center();

  input {
    margin-right: 0.5rem;
    cursor: pointer;
  }
}

.recorder {
  button {
    margin: 0 0.15rem;
    padding: 0.5rem 1rem;
    background-color: $primary;
    color: $light;
    border: none;
    outline: none;
    border-radius: 2px;
    box-shadow: 0 1px 2px rgba(0, 0, 0, 0.3);
    cursor: pointer;
    user-select: none;

    &.pause {
      background-color: $warning;
    }

    &.resume {
      background-color: $success;
    }

    &.stop {
      background-color: $danger;
    }
  }
}

video {
  max-width: 320px;
  border-radius: 4px;
}

audio {
  margin: 0.25rem 0;
}
