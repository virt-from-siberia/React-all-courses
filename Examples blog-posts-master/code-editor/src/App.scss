$primary: #0275d8;
$success: #5cb85c;
$warning: #f0ad4e;
$danger: #d9534f;
$light: #f7f7f7;
$dark: #292b2c;

@mixin reset($font-family, $font-size, $color) {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  @if $font-family {
    font-family: $font-family;
  }
  @if $font-size {
    font-size: $font-size;
  }
  @if $color {
    color: $color;
  }
}

@mixin flex-center($column: false) {
  display: flex;
  justify-content: center;
  align-items: center;

  @if $column {
    & {
      flex-direction: column;
    }
  }
}

@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@200;400;600&display=swap');

*:not(.react-codemirror2 *) {
  @include reset('Montserrat', 1rem, $dark);
}

.app {
  @include flex-center(true);
  margin: 0 auto;
  max-width: 600px;

  h1 {
    margin: 1rem 0;
    font-size: 1.6rem;
    text-align: center;
  }

  .tabs {
    button {
      margin: 0.5rem;
      padding: 0.5rem;
      background: none;
      border: none;
      outline: none;
      border-radius: 6px;
      transition: 0.4s;
      cursor: pointer;
      user-select: none;

      &:hover,
      &.current {
        background: $dark;
        color: $light;
      }
    }
  }

  .theme-selector {
    margin: 0.75rem 0;
    text-align: center;

    select {
      border-radius: 4px;
    }
  }

  .code-editor {
    width: 100%;

    .CodeMirror-wrap {
      border-radius: 4px;
      padding: 0.25rem;
    }
  }

  .code-executor {
    width: 100%;
  }

  .btn {
    margin: 0.75rem;
    padding: 0.25rem 0.75rem;
    background: $primary;
    border: none;
    border-radius: 4px;
    outline: none;
    color: $light;
    box-shadow: 0 1px 2px rgba(0, 0, 0, 0.5);
    cursor: pointer;
    user-select: none;
    transition: 0.3s;

    &:active {
      box-shadow: none;
    }

    &.run {
      background: $success;
    }
  }

  iframe {
    padding: 0.25rem;
    width: 100%;
    border: 1px dashed $dark;
    border-radius: 4px;
  }
}
