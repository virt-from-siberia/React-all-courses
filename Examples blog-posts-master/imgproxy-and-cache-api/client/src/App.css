.App {
  padding: 1rem;
}

h1 {
  text-align: center;
}

.images {
  display: flex;
  flex-wrap: wrap;
}

img {
  display: block;
  max-width: 100%;
}
