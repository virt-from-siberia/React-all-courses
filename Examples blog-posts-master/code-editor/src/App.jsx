import './App.scss'
import { useState } from 'react'
import { Tabs } from './components/Tabs'
import { CodeEditor } from './components/CodeEditor'
import { CodeExecutor } from './components/CodeExecutor'

const initialHTML = '<h1>hi</h1>'
const initialCSS = `
h1 {
  color: green;
}
`
const initialJavaScript = `
document.querySelector("h1").addEventListener('click', function () {
  this.textContent = "bye"
  this.style.color = "red"
}, { once: true })
`

export default function App() {
  const [mode, setMode] = useState('html')
  const [html, setHtml] = useState(initialHTML)
  const [css, setCss] = useState(initialCSS.trim())
  const [js, setJs] = useState(initialJavaScript.trim())
  const [srcDoc, setSrcDoc] = useState('')

  const runCode = () => {
    setSrcDoc(
      `<html>
        <style>${css}</style>
        <body>${html}</body>
        <script>${js}</script>
      </html>`
    )
  }

  const propsByMode = {
    html: {
      mode: 'xml',
      value: html,
      setValue: setHtml
    },
    css: {
      mode: 'css',
      value: css,
      setValue: setCss
    },
    js: {
      mode: 'javascript',
      value: js,
      setValue: setJs
    }
  }

  return (
    <div className='app'>
      <h1>React Code Editor</h1>
      <Tabs mode={mode} setMode={setMode} />
      <CodeEditor {...propsByMode[mode]} />
      <CodeExecutor srcDoc={srcDoc} runCode={runCode} />
    </div>
  )
}
