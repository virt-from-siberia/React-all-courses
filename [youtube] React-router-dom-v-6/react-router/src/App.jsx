import { Routes, Route, Link } from "react-router-dom";

import { AboutPage } from "./pages/AboutPage";
import { BlogPage } from "./pages/BlogPage";
import { HomePage } from "./pages/HomePage";
import { NotPoundPage } from "./pages/NotPoundPage";

import { Layout } from "./components/Layout";

import "./App.css";

function App() {
  return (
    <div className="App">
      <Routes>
        <Route path="" element={<Layout />}>
          <Route index element={<HomePage />} />
          <Route path="about" element={<AboutPage />} />
          <Route path="posts" element={<BlogPage />} />
          <Route path="*" element={<NotPoundPage />} />
        </Route>
      </Routes>
    </div>
  );
}

export default App;
