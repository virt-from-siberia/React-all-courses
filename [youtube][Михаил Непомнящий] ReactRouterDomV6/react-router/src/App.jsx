import { Routes, Route, Navigate } from "react-router-dom";

import { AboutPage } from "./pages/AboutPage";
import { BlogPage } from "./pages/BlogPage";
import { HomePage } from "./pages/HomePage";
import { NotPoundPage } from "./pages/NotPoundPage";
import { SinglePage } from "./pages/SinglePage";
import { CreatePost } from "./pages/CreatePost";
import { EditPost } from "./pages/EditPost";
import { LoginPage } from "./pages/LoginPage";

import { Layout } from "./components/Layout";

import { RequireAuth } from "./hoc/RequireAuth";
import { AuthProvider } from "./hoc/AuthProvider";

import "./App.css";

function App() {
  return (
    <div className="App">
      <AuthProvider>
        <Routes>
          <Route path="" element={<Layout />}>
            <Route index element={<HomePage />} />

            <Route path="about/*" element={<AboutPage />}>
              <Route path="contacts" element={<p>our contacts</p>} />
              <Route path="team" element={<p>our team</p>} />
            </Route>

            <Route path="about-us" element={<Navigate to="/about" replace />} />
            <Route path="posts" element={<BlogPage />} />
            <Route path="posts/:id" element={<SinglePage />} />
            <Route path="posts/:id/edit" element={<EditPost />} />
            <Route
              path="posts/new"
              element={
                <RequireAuth>
                  <CreatePost />
                </RequireAuth>
              }
            />
            <Route path="login" element={<LoginPage />} />
            <Route path="*" element={<NotPoundPage />} />
          </Route>
        </Routes>
      </AuthProvider>
    </div>
  );
}

export default App;
