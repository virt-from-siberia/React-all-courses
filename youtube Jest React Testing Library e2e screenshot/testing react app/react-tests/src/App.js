import "./App.css";

import { useState, useEffect } from "react";
import logo from "./logo.svg";

function App() {
  const [data, setData] = useState();

  useEffect(() => {
    setTimeout(() => {
      setData(100);
    }, 100);
  }, []);

  return (
    <div>
      <h1>Hello world</h1>
      {data && <div style={{ color: "red" }}>data</div>}
      <button>click me</button>
      <input placeholder="input value..." type="text" />
    </div>
  );
}

export default App;
