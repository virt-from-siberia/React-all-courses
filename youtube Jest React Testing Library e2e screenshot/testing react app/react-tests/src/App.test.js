import { render, screen } from "@testing-library/react";
import App from "./App";

describe("test App", () => {
  test("renders learn react link", async () => {
    render(<App />);

    screen.debug();

    const helloWorldElement = await screen.findByText(/data/i);
    expect(helloWorldElement).toBeInTheDocument();
    expect(helloWorldElement).toHaveStyle({ color: "red" });

    screen.debug();
    // const helloWorldElement = screen.queryByText(/Hello2/i);
    // expect(helloWorldElement).toBeNull();

    // const helloWorldElement = screen.getByText(/Hello world/i);
    // const btn = screen.getByRole("button");
    // const input = screen.getByPlaceholderText(/input value/i);

    // expect(helloWorldElement).toBeInTheDocument();
    // expect(btn).toBeInTheDocument();
    // expect(input).toBeInTheDocument();
    // expect(input).toMatchSnapshot();

    // screen.debug();
  });
});
