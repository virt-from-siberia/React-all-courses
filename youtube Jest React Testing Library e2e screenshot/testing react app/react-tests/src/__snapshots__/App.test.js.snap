// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;

exports[`test App renders learn react link 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;
